rollup/types.go

```diff
@@ -331,7 +335,15 @@ func (c *CTCTransaction) Decode(b []byte) error {
 			return fmt.Errorf("Cannot decode EIP155 ctc tx %x: %w", b, err)
 		}
 		c.tx = &tx
+	case CTCTransactionTypeEthSign:
+		tx := CTCTxEthSign{}
+		err := tx.Decode(b[1:])
+		if err != nil {
+			return fmt.Errorf("Cannot decode EthSign ctc tx %x: %w", b, err)
+		}
+		c.tx = &tx
 	}
+
```

Comment:
Is the decoding different for `eth_sign` transactions?

Reply:
The decoding is exactly the same, but no inheritance in go. I did mention a solution to deduplicating things: https://github.com/ethereum-optimism/go-ethereum/pull/111#discussion_r527147848